src/Lib/Db.hs

```diff
@@ -0,0 +1,54 @@
+{-# LANGUAGE FlexibleContexts #-}
+
+module Lib.Db
+       ( queryPG
+       , queryPG_
+       , executePG
+       , executePG_
+       , executeManyPG
+       ) where
+
+import           Lib.App.Env                (AppEnv (..))
+
+import qualified Data.Pool                  as Pool
+import qualified Database.PostgreSQL.Simple as PG
+
+-- Query the database with a given query and args and expect a list of
+-- rows in return
+queryPG :: (MonadReader AppEnv m, MonadIO m, PG.ToRow q, PG.FromRow r) =>
+  PG.Query -> q -> m [r]
```

Comment:
Okay, after you moved all of these SQL functions into separate module, it's time to format those functions according to how it's done in CDMP:

* https://github.com/Holmusk/CDMP/blob/c9f1ba8513adf13720dbf5a73ed4b48afdd2c1aa/backend/src/CDMP/Db.hs#L20-L35

Specifically:

1. Write type on single line.
2. Remove PG suffix from functions.
3. Change `import qualified Database.PostgreSQL.Simple as PG` to `import qualified Database.PostgreSQL.Simple as SQL`.
4. Move `(MonadReader AppEnv m, MonadIO m)` constraint into `WithDbPool` constraint alias.
5. Replace `(MonadReader AppEnv m, MonadIO m)` in every function in whole package with `WithDbPool m`.
6. Move `asSingleRow` function in this module.
7. Implement `perform` function and rewrite every SQL function in this module to use `perform`